test/unit/ObjectMakrTests/processFunctions.js

```diff
@@ -0,0 +1,90 @@
+define(["mocks"], function (mocks) {
+    return function () {
+        var expect = require("chai").expect;
+
+        it("processes the correct number of functions", function () {
+            // Arrange
+            var objectMaker = mocks.mockObjectMakr({
+                inheritance: {
+                    Organism: {
+                        Animal: {
+                            Bird:{
+                                Penguin: {}
+                            },
+                            Mammal: {
+                                Dog: {}
+                            }
+                        },
+                        Plant: {}
+                    }
+                },
+                properties: {
```

Comment:
Are you saying this for `process*` tests and `proliferate` as well and to not use the default constructor in `mocks`?